### PR TITLE
BaSP-TG-100: Projects redux - Create, update and delete

### PR DESCRIPTION
**Pull request by GonzaloAsencio**
**Create project**
Create constants to create an project
Create actions to create an project
Create thunk to create project
Complete projects reducer
Connect form component with store to create an project

**Updated project**
Create constants to update an project
Create actions to update an project
Create thunk to update project
Complete update reducer
Connect form component with store to update an project

**Delete project**
Create constants to delete an project
Create actions to delete an project
Create thunk to delete project
Complete delete reducer
Connect form component with store to delete an project

### CODE REVIEW

File: src/redux/projects/actions.js [hunk 1]
```diff
@@ -1,4 +1,17 @@
...
+  GET_PROJECTS_ERROR,
+  POST_PROJECT_PENDING,
+  POST_PROJECT_SUCCESS,
+  POST_PROJECT_ERROR,
+  DELETE_PROJECT_PENDING,
+  DELETE_PROJECT_ERROR,
+  DELETE_PROJECT_SUCCESS,
+  PUT_PROJECT_PENDING,
+  PUT_PROJECT_ERROR,
+  PUT_PROJECT_SUCCESS
```
Comment (fede230298) [inline src/redux/projects/actions.js:13]: Adjust order to be 
GET
POST
PUT
DELETE

File: src/redux/projects/actions.js [hunk 2]
```diff
@@ -19,3 +32,63 @@ export const getProjectsError = (error) => {
...
+  };
+};
+
+export const deleteProjectError = (error) => {
+  return {
+    type: DELETE_PROJECT_ERROR,
+    payload: error
+  };
+};
+
```
Comment (fede230298) [inline src/redux/projects/actions.js:55]: Same as above, lets mantain the order
GET
POST
PUT 
DELETE

File: src/redux/projects/constant.js [hunk 3]
```diff
@@ -1,3 +1,12 @@
 export const GET_PROJECTS_PENDING = 'GET_PROJECTS_PENDING';
 export const GET_PROJECTS_SUCCESS = 'GET_PROJECTS_SUCCESS';
 export const GET_PROJECTS_ERROR = 'GET_PROJECTS_ERROR';
+export const DELETE_PROJECT_PENDING = 'DELETE_PROJECT_PENDING';
+export const DELETE_PROJECT_SUCCESS = 'DELETE_PROJECT_SUCCESS';
+export const DELETE_PROJECT_ERROR = 'DELETE_PROJECT_ERROR';
```
Comment (fede230298) [inline src/redux/projects/constant.js:6]: Same as above, put in order

File: src/redux/projects/reducer.js [hunk 4]
```diff
@@ -1,8 +1,22 @@
...
+import {
+  GET_PROJECTS_PENDING,
+  GET_PROJECTS_SUCCESS,
+  GET_PROJECTS_ERROR,
+  DELETE_PROJECT_PENDING,
+  DELETE_PROJECT_SUCCESS,
+  DELETE_PROJECT_ERROR,
+  POST_PROJECT_PENDING,
+  POST_PROJECT_SUCCESS,
+  POST_PROJECT_ERROR,
```
Comment (fede230298) [inline src/redux/projects/reducer.js:10]: Order

File: src/redux/projects/reducer.js [hunk 5]
```diff
@@ -27,6 +41,67 @@ const reducer = (state = INITIAL_STATE, action) => {
...
+        isLoading: false,
+        error: ''
+      };
+    case DELETE_PROJECT_ERROR:
+      return {
+        ...state,
+        isLoading: false,
+        error: action.payload,
+        list: []
+      };
```
Comment (fede230298) [inline src/redux/projects/reducer.js:62]: Order

File: src/redux/projects/thunks.js [hunk 6]
```diff
@@ -17,3 +27,73 @@ export const getProjects = () => {
...
+        } else {
+          dispatch(deleteProjectSuccess(id));
+        }
+      })
+      .catch((error) => {
+        dispatch(deleteProjectError(error.toString()));
+      });
+  };
+};
+
```
Comment (fede230298) [inline src/redux/projects/thunks.js:52]: Order

File: src/redux/projects/actions.js (hunk 1 above)
Comment (GonzaloAsencio) [reply src/redux/projects/actions.js]: Done

File: src/redux/projects/actions.js (hunk 2 above)
Comment (GonzaloAsencio) [reply src/redux/projects/actions.js]: Done

File: src/redux/projects/constant.js (hunk 3 above)
Comment (GonzaloAsencio) [reply src/redux/projects/constant.js]: Done

File: src/redux/projects/reducer.js (hunk 4 above)
Comment (GonzaloAsencio) [reply src/redux/projects/reducer.js]: Done

File: src/redux/projects/reducer.js (hunk 5 above)
Comment (GonzaloAsencio) [reply src/redux/projects/reducer.js]: Done

File: src/redux/projects/thunks.js (hunk 6 above)
Comment (GonzaloAsencio) [reply src/redux/projects/thunks.js]: Done

**Review by ProDIGGY-vinilos — APPROVED** [review]

File: src/Components/Projects/Form/index.js [hunk 7]
```diff
@@ -50,35 +59,27 @@ const Project = () => {
 
   useEffect(async () => {
```
Comment (JuliMonte5) [inline src/Components/Projects/Form/index.js:60]: Don't use async here

File: src/Components/Projects/Form/index.js [hunk 8]
```diff
@@ -50,35 +59,27 @@ const Project = () => {
...
+          }
+        ]
+      });
     } else {
       setIsFetched(false);
     }
-    setIsLoading(false);
   }, []);
 
   useEffect(async () => {
```
Comment (JuliMonte5) [inline src/Components/Projects/Form/index.js:85]: Same as 51, fetch actions should be done with dispatch hook

File: src/redux/projects/reducer.js [hunk 9]
```diff
@@ -27,6 +41,67 @@ const reducer = (state = INITIAL_STATE, action) => {
...
+    case PUT_PROJECT_PENDING:
+      return {
+        ...state,
+        isLoading: true,
+        modalMessage: ''
+      };
+    case PUT_PROJECT_SUCCESS:
+      return {
+        ...state,
+        list: [...state.list, action.payload.data],
```
Comment (fede230298) [inline src/redux/projects/reducer.js:74]: You are adding a new element to the list. Lets use .map to update the list

File: src/redux/projects/reducer.js [hunk 10]
```diff
@@ -1,8 +1,22 @@
...
+  POST_PROJECT_ERROR,
+  PUT_PROJECT_PENDING,
+  PUT_PROJECT_SUCCESS,
+  PUT_PROJECT_ERROR,
+  DELETE_PROJECT_PENDING,
+  DELETE_PROJECT_SUCCESS,
+  DELETE_PROJECT_ERROR
+} from './constant';
 
 const INITIAL_STATE = {
```
Comment (Ignacio-Sormani) [inline src/redux/projects/reducer.js:16]: Only use list, isLoading and error in redux

File: src/redux/projects/reducer.js (hunk 9 above)
Comment (Ignacio-Sormani) [inline src/redux/projects/reducer.js:74]: Use map function to update list

File: src/Components/Projects/Form/index.js [hunk 11]
```diff
@@ -7,8 +7,11 @@ import MessageModal from '../../Shared/Modal/MessageModal';
 import styles from './form.module.css';
 import Button from '../../Shared/Button/Button';
 import InputField from '../../Shared/Input/input';
+import { useSelector, useDispatch } from 'react-redux';
+import { postProject, putProject } from '../../../redux/projects/thunks';
 
 const Project = () => {
+  let formTtitle = 'Tittle';
```
[the remark this answers is not in the document]
Comment (GonzaloAsencio) [reply src/Components/Projects/Form/index.js:14]: My bad... Done

File: src/Components/Projects/Form/index.js (hunk 7 above)
Comment (GonzaloAsencio) [reply src/Components/Projects/Form/index.js:60]: Done

File: src/redux/projects/reducer.js [hunk 12]
```diff
@@ -27,6 +41,67 @@ const reducer = (state = INITIAL_STATE, action) => {
...
+      };
+    case DELETE_PROJECT_PENDING:
+      return {
+        ...state,
+        isLoading: true
+      };
+    case DELETE_PROJECT_SUCCESS:
+      return {
+        ...state,
+        list: [...state.list.filter((project) => project._id !== action.payload)],
```
Comment (ProDIGGY-vinilos) [inline src/redux/projects/reducer.js:94]: Let's update to use .map

**Review by fede230298 — APPROVED** [review]

File: src/Components/Projects/Form/index.js [hunk 13]
```diff
@@ -7,8 +7,12 @@ import MessageModal from '../../Shared/Modal/MessageModal';
 import styles from './form.module.css';
 import Button from '../../Shared/Button/Button';
 import InputField from '../../Shared/Input/input';
+import { useSelector, useDispatch } from 'react-redux';
+import { postProject, putProject } from '../../../redux/projects/thunks';
+import getEmployees from '../../../redux/employees/thunks';
 
 const Project = () => {
+  let formTitle = 'Title';
```
Comment (fede230298) [inline src/Components/Projects/Form/index.js:15]: What is the purpose of this?

**Pull request merged**